src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './FindMovie.scss';
+import cn from 'classnames';
+import { Movie } from '../../types/Movie';
+import { getMovie } from '../../api';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  addMovie: (movie: Movie) => void;
+};
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+  const [movieNotFound, setMovieNotFound] = useState(false);
+  const [isLoad, setIsLoad] = useState(false);
+  const buttonDisabled = query.length > 0;
+
+  useEffect(() => {
+    if (movieNotFound) {
+      setMovieNotFound(false);
+    }
+  }, [query]);
+
+  const handleFormSubmit = async () => {
+    setIsLoad(true);
+    const movieData: MovieData | ResponseError = await getMovie(query);
+
+    if ('Title' in movieData) {
+      if (movieData.Poster === 'N/A') {
+        // eslint-disable-next-line max-len
+        movieData.Poster = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+      }
+
+      const newMovie: Movie = {
+        title: movieData.Title,
+        description: movieData.Plot,
+        imgUrl: movieData.Poster,
+        imdbUrl: `https://www.imdb.com/title/${movieData.imdbID}`,
+        imdbId: movieData.imdbID,
+      };
+
+      setFoundMovie(newMovie);
+    } else {
+      setMovieNotFound(true);
+    }
+  };
```

Comment:
1. try catch
2. better move to helper but no critical

Reply:
Thanks, fixed